irods/manager/user_manager.py

```diff
@@ -187,25 +207,49 @@ def get(self, name, user_zone=""):
             raise UserGroupDoesNotExist()
         return iRODSUserGroup(self, result)
 
-    def create(self, name, user_type='rodsgroup', user_zone="", auth_str="", group_admin=False):
-        if group_admin:
-            message_body = UserAdminRequest(
+    def _api_info(self, group_admin_flag):
+        """
+        If group_admin_flag is:         then use UserAdminRequest API:
+        ---------------------------     ------------------------------
+        True                            always
+        False (user_groups default)     never
+        None  (groups default)          when user type is groupadmin
+        """
+
+        sess = self.sess
+        if group_admin_flag or (group_admin_flag is not False and sess.users.get(sess.username).type == 'groupadmin'):
+            return (UserAdminRequest, 'USER_ADMIN_AN')
+        return (GeneralAdminRequest, 'GENERAL_ADMIN_AN')
+
+    def create(self, name,
+               user_type = 'rodsgroup',
+               user_zone = "",
+               auth_str = "",
+               group_admin = False, **options):
+
+        if not options.pop('suppress_deprecation_warning', False):
+            warnings.warn('Use of session.user_groups is deprecated in v1.1.7 - prefer session.groups',
+                          DeprecationWarning, stacklevel = 2)
+
+        (MessageClass, api_key) = self._api_info(group_admin)
+        api_to_use = api_number[api_key]
+
+        if MessageClass is UserAdminRequest:
+            message_body = MessageClass(
                 "mkgroup",
                 name,
                 user_type,
                 user_zone
             )
-            api_to_use = api_number['USER_ADMIN_AN']
         else:
-            message_body = GeneralAdminRequest(
+            message_body = MessageClass(
                 "add",
                 "user",
                 name,
                 user_type,
-                "",
+                user_zone, # "", # TODO test this # new issue?
```

Comment:
I think making a test for this might be more like testing the API rather than the client, but it could be worth writing a test here as well